pages/index.tsx

```diff
@@ -1,17 +1,37 @@
 import { loadProfiles } from "../util/load-profiles";
 import Head from "next/head";
-import OverviewCard from "../components/overview-card";
 import { ProfileType } from "../types/ProfileType";
+import SearchFilter from "../components/SearchFilter";
+import { useCallback, useEffect, useState } from "react";
+import OverviewCard from "../components/Profile";
 
 export default function Home({ profiles }: { profiles: ProfileType[] }) {
+  const [searchString, setSearchString] = useState("")
+  const [filteredProfiles, setFilteredProfiles] = useState<ProfileType[]>([])
+
+  const filterBySearchString = useCallback(() => {
+    if(searchString !== ""){
+      const filteredProfiles = profiles.filter(profile => profile.name.toLowerCase().includes(searchString.toLowerCase()) || profile.location.toLowerCase().includes(searchString.toLowerCase()))
+      setFilteredProfiles(filteredProfiles)
+    } else {
+      setFilteredProfiles([])
+    }
+  },[profiles, searchString])
```

Comment:
I think you could refactor this that way you won't have to do `profiles.map` : `filteredProfiles.map`

set the inital state of filterProfiles to `profiles`

in the else part of `filterBySearchString` you could just `setFilteredProfiles(profiles)`

and then just have `filteredProfiles.map`